rktl_sim/nodes/simulator

```diff
@@ -245,33 +218,71 @@ class Simulator(object):
         if req.name in self.car_pose_pubs:
             self.car_pose_pubs[req.name].unregister()
             del self.car_pose_pubs[req.name]
-        
-        # Delete the car's body from pybullet
+
+        # Delete cars body from pybullet.
         res = self.sim.delete_car(self.car_ids[req.name])
         del self.car_ids[req.name]
-        
+
+        self.reset_cb()
         self.reset_lock.release()
+
         return DeleteCarResponse(res)
 
-    def reset_cb(self, _):
-        """Resets simulator"""
+    def reset_cb(self, reset_sim=True):
```

Comment:
Can't think of a scenario where this function doesn't reset the sim. We should maybe update the docstring as well. Also this is a ROS service callback, which requires we have an additional argument for the request. It's an empty request, hence the underscore, but it's still required.

Reply:
Ok, I think I know what needs to be done, but before I do it, I want to check that this is the right way to go. Instead of traversing from simulator to sim to car/ball objects, I should use rospy.get_param in the car/ball objects to constantly refer to parameters in the rospy. That way when I call the reset_cb function in the simulator, using the rospy.set_param, I will change the parameters in ropsy, and hence when the car and ball objects grab from ropsy via rospy.get_param, their own parameters will get reset.

I found an example of this code where something like this happens:
`car_param = [
  {
    "name": "car0",
    "init_pose": {
      "pos": [
        init_car_odom.pose.pose.position.x,
        init_car_odom.pose.pose.position.y,
        0.06,
      ],
      "orient": [
        0.0, 0.0, car_euler[2]
      ]
    }
  }
]

rospy.set_param(f'{sim_name}/cars', car_param)`

This will be called in the reset_cb function instead of passing parameters to sim.py, car.py and the ball object and physically changing the old parameters in each object. Is this correct?

Because if it is, it will save a bunch of time as by using rospy to link the simulator to the sim /car and ball objects, it will clean up a lot of code that currently does the same functionality. It will also make it easier to follow what happens in the sim.py and the car/ball objects as there will not be a large trail of variables heading from the simulator to the sim to the ball objects. It will also be very clean.

Reply:
We resolved this over discord, let me know if any questions remain